src/DateRangePicker.ts

```diff
@@ -1,26 +1,423 @@
+    this._leftSliderX = Math.max(toSet, this.sliderWidth);
+  }
+
+  setRightSlider(newX: number): void {
+    const toSet = Math.max(newX, this._leftSliderX);
+    this._rightSliderX = Math.min(toSet, this.width - this.sliderWidth);
+  }
+
+  translatePositionToDate(x: number): string {
+    const milliseconds =
+      ((x - this.sliderWidth) * this.dateRange()) / this._histWidth;
+    const date = dayjs(this._minDate + milliseconds);
+    return date.format(this.dateFormat);
+  }
+
+  translateDateToPosition(date: string): number {
+    const milliseconds: number | undefined = dayjs(date).valueOf();
+    if (!milliseconds) {
+      return DATE_PARSING_ERROR;
+    }
+    // translate where we are within the date range into what the new x-position
+    // of the slider should be
+    return (
+      this.sliderWidth +
+      ((milliseconds - this._minDate) * this._histWidth) / this.dateRange()
+    );
+  }
+
+  handleDateInput(e: InputEvent): void {
+    const target = e.currentTarget as HTMLInputElement;
+    const newX = this.translateDateToPosition(target.value);
+    if ((target.id as InputIds) === 'date-min') {
+      if (newX !== DATE_PARSING_ERROR) {
+        this.setLeftSlider(newX);
+      }
+      target.value = this.translatePositionToDate(this._leftSliderX);
+    } else {
+      if (newX !== DATE_PARSING_ERROR) {
+        this.setRightSlider(newX);
+      }
+      target.value = this.translatePositionToDate(this._rightSliderX);
+    }
+  }
+
+  renderSlider(id: SliderIds): SVGTemplateResult {
+    const x = id === 'slider-min' ? this._leftSliderX : this._rightSliderX;
+    const k = id === 'slider-min' ? 1 : -1;
+    const c = SLIDER_CORNER_SIZE;
+    const sliderShape =
+      id === 'slider-min'
+        ? `
+            M${x},0
+            h-${this.sliderWidth - c}
+            q-${c},0 -${c},${c}
+            v${this.height - c * 2}
+            q0,${c} ${c},${c}
+            h${this.sliderWidth - c}
+          `
+        : `
+            M${x},0
+            h${this.sliderWidth - c}
+            q${c},0 ${c},${c}
+            v${this.height - c * 2}
+            q0,${c} -${c},${c}
+            h-${this.sliderWidth - c}
+          `;
+    return svg`
+    <svg
+      id="${id}"
+      @pointerdown="${this.drag}"
+      @pointerup="${this.drop}"
+    >
+      <path d="${sliderShape} z" fill="${sliderFill}" />
+      <rect
+        x="${x - this.sliderWidth * k + this.sliderWidth * 0.4 * k}"
+        y="${this.height / 3}"
+        width="1"
+        height="${this.height / 3}"
+        fill="white"
+      />
+      <rect
+        x="${x - this.sliderWidth * k + this.sliderWidth * 0.6 * k}"
+        y="${this.height / 3}"
+        width="1"
+        height="${this.height / 3}"
+        fill="white"
+      />
+    </svg>
+    `;
+  }
+
+  barIncluded(x: number): boolean {
+    return x >= this._leftSliderX && x <= this._rightSliderX;
+  }
+
+  renderSelectedRange(): SVGTemplateResult {
```

Comment:
_Suggestion_ Make this a getter since it's just using internal properties to render:
```
private get selectedRangeTemplate(): SVGTemplateResult {}

...

${this.selectedRangeTemplate}
```